enhancements/network/user-defined-network-segmentation.md

```diff
@@ -565,6 +566,13 @@ type ClusterUserDefinedNetwork struct {
 
 // ClusterUserDefinedNetwork defines the desired state of ClusterUserDefinedNetwork.
 type ClusterUserDefinedNetwork struct {
+    // VRF is an optional VRF device name to use for the network. It has to be less than
+    // 16 characters long and unique. If omitted, the ClusterUserDefinedNetwork name
+    // will be used as long as it is less than 16 characters long. Otherwise an internal
+    // name will be generated.
+    // +optional
```

Comment:
> Why does the VRF name has to be configured implicitly if not specified in the spec?

Note that that is already happening. We currently use VRF names as `mp<network-id>-udn-vrf`

> Why does the VRF name has to be configured implicitly if not specified in the spec? Can we avoid forcing VRF name for CUDNs who doesnt specify it?
> 
> Please consider simplifying the name generation as much possible, it will make so it easier to work with and more predictable. I would drop using the CUDN metadata.name, leaving two options: the spesifed name in the spec or generated one.
> 

Note that currently we are using a generated VRF name as `mp<network-id>-udn-vrf` which is not predictable and which is precisely what this proposal is trying to address.

Using `metadata.name` is what we should be doing, except for the fact that is not limited to 15 characters. Because of this, we offer the proposed optional extra field.

Generating a name is a fallback we do need just because of that same reason: `metadata.name` is not limited to 15 characters.

The only simplification we can do here is making the field required. But that is not backward compatible and will potentially lead to just replicating the `metadata.name` value in most cases.